src/components/Contacts/ContactFilters/ContactFilters.tsx

```diff
@@ -1,35 +1,211 @@
+        <Slide
+          in={selectedGroup == null}
+          direction="right"
+          appear={false}
+          mountOnEnter
+          unmountOnExit
+        >
+          <div>
+            <FilterHeader>
+              <Typography variant="h6">
+                {Object.keys(selectedFilters).length > 0
+                  ? t('Filter ({{count}})', {
+                      count: Object.keys(selectedFilters).length,
+                    })
+                  : t('Filter')}
+              </Typography>
+              <LinkButton
+                color="primary"
+                style={{ marginInlineStart: theme.spacing(-1) }}
+                disabled={Object.keys(selectedFilters).length == 0}
+                onClick={() => alert('TODO')}
+              >
+                {t('Save')}
+              </LinkButton>
+              <LinkButton
+                color="primary"
+                style={{ marginInlineStart: theme.spacing(2) }}
+                disabled={Object.keys(selectedFilters).length == 0}
+                onClick={() => setSelectedFilters({})}
+              >
+                {t('Clear All')}
+              </LinkButton>
+            </FilterHeader>
+            <FilterList dense>
+              {error && (
+                <ListItem data-testid="ErrorState">
+                  <ListItemText
+                    primary={error.toString()}
+                    primaryTypographyProps={{
+                      variant: 'subtitle1',
+                      color: 'error',
+                    }}
+                  />
+                </ListItem>
+              )}
+              {loading ? (
+                <ListItem data-testid="LoadingState">
+                  <CircularProgress />
+                </ListItem>
+              ) : data?.contactFilters?.length == 0 ? (
+                <ListItem data-testid="NoFiltersState">
+                  <ListItemText
+                    primary={t('No Contact Filters Found')}
+                    primaryTypographyProps={{ variant: 'subtitle1' }}
+                  />
+                </ListItem>
+              ) : (
+                <>
+                  {data?.contactFilters?.map((group: ContactFilterGroup) => (
+                    <Collapse
+                      key={group.id}
+                      in={showAll || isGroupVisible(group)}
+                      data-testid="FilterGroup"
+                    >
+                      <ListItem button onClick={() => showGroup(group)}>
+                        <ListItemText
+                          primary={group.title}
+                          primaryTypographyProps={{ variant: 'subtitle1' }}
+                        />
+                        <ArrowForwardIos fontSize="small" color="disabled" />
+                      </ListItem>
+                    </Collapse>
+                  ))}
+                  {data?.contactFilters?.some(
+                    (g: ContactFilterGroup) => !isGroupVisible(g),
```

Comment:
```suggestion
                    (g) => !isGroupVisible(g),
```

Reply:
this causes a not assignable type error for `isGroupVisible(g)`

Reply:
nm, this goes away once I use the `FilterGroup = ContactFiltersQuery['contactFilters']` type alias